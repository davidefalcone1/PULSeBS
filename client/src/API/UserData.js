class UserData{
    constructor(id, personId, fullName, email) {
        if (id) 
            this.id = id;
        this.personId = personId;
        this.fullName = fullName;
        this.email = email;
    }

    static fromJson(json){
        const temp =  Object.assign(new UserData(), json);
        return temp;
    }
}

export default UserData;