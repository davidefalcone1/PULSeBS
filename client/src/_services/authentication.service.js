import { BehaviorSubject } from 'rxjs';


const currentUserSubject = new BehaviorSubject(JSON.parse(localStorage.getItem('user')));

export const authenticationService = {
    login,
    logout,
    user: currentUserSubject.asObservable(),
    get currentUserValue () { return currentUserSubject.value }
};

function handleResponse(response) {
    return response.text().then(text => {
        const data = text && JSON.parse(text);
        if (!response.ok) {
            if ([401, 403].indexOf(response.status) !== -1) {
                // auto logout if 401 Unauthorized or 403 Forbidden response returned from api
                authenticationService.logout();
            }

            const error = (data && data.message) || response.statusText;
            return Promise.reject(error);
        }

        return data;
    });
}

function login(username, password) {
    const requestOptions = {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({ username, password })
    };
    return new Promise(async function (resolve, reject) {
        try {
            const response = await fetch('/users/authenticate', {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json',
                },
                body: JSON.stringify({ username, password }),
            });
            const user = await response.json().then(handleResponse);
            if (response.ok) {
                localStorage.setItem('user', JSON.stringify(user));
                currentUserSubject.next(user);
                resolve(user);
            }
            else
                reject();
        }
        catch (e) {
            reject();
        }
    });

}
function logout() {

    const requestOptions = {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
    };
    localStorage.removeItem('user');
    currentUserSubject.next(null);
    return fetch('/logout',requestOptions);
}