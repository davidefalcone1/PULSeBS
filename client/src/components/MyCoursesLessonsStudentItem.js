import React from 'react';
import ListGroup from 'react-bootstrap/ListGroup';
import Button from 'react-bootstrap/Button'

function myCoursesLessonsStudentItem (props) {
    return (
        <ListGroup.Item id = {"student-" + props.student.personId}>
            <div className="d-flex w-100 pt-3 justify-content-between no-gutters">
                <PersonIdField id = {props.student.personId}/>
                <FullNameField id = {props.student.personId} fullName = {props.student.fullName}/>
                <EmailField id = {props.student.personId} email = {props.student.email}/>
                <SelectField id = {props.student.personId} 
                    booking = {props.booking} setStudentAsPresent = {props.setStudentAsPresent}/>
            </div>
        </ListGroup.Item>
    );
}

function PersonIdField(props){
    return(
        <div className="col-sm-2">
            <p id={"idOfStudent_" + props.id}>
                {props.id}
            </p>
        </div>
    );
}

function FullNameField(props){
    return(
        <div className="col-sm-3">
            <p id={"fullNameOfStudent_" + props.id}>
                {props.fullName}
            </p>
        </div>
    );
}

function EmailField(props){
    return(
        <div className="col-sm-4">
            <p id={"emailOfStudent" + props.id}>
                {props.email}
            </p>
        </div>
    );
}

function SelectField(props){
    return(
        <div className="col-sm-3">
            {!props.booking.attended &&  
                <Button variant="success" onClick={(event) => {
                    event.preventDefault();
                    props.setStudentAsPresent(props.booking.scheduleId, props.booking.studentId);
                }} id={"selectFieldOfLesson" + props.booking.scheduleId + " - " + props.booking.studentId}>
                    Set as Present
                </Button>
            }
        </div>
    );
}

export default myCoursesLessonsStudentItem;