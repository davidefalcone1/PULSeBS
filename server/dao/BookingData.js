class BookingData {
    constructor(id, scheduleId, studentId, status, attended) {
        if (id)
            this.id = id;
        this.scheduleId = scheduleId;
        this.studentId = studentId;
        this.status = status;
        this.attended = attended;
    }

    static fromJson(json) {
        const temp = Object.assign(new BookingData(), json);
        return temp;
    }
}

module.exports = BookingData;