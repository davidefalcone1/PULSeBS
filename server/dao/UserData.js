class UserData{
    constructor(id, personId, fullName, email) {
        if (id) 
            this.id = id;
        this.personId = personId;
        this.fullName = fullName;
        this.email = email;
    }

}

module.exports = UserData;