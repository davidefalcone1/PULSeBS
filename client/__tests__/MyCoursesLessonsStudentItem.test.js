import React from 'react';
import renderer from 'react-test-renderer';
import { AuthContext } from '../src/_services/AuthContext';
import MyCoursesLessonsStudentItem from '../src/components/MyCoursesLessonsStudentItem';
import { BrowserRouter } from 'react-router-dom';
import UserData from '../src/API/UserData';
import BookingData from '../src/API/BookingData';

describe('MyCoursesLessonsStudentItem', ()=>{
    test('Renders correctly', ()=>{
        const component = renderer.create(
            <BrowserRouter>
                <AuthContext.Provider value={{user: new UserData(1, '12345', 'Davide Falcone', 'davide.falcone@polito.it')}}>
                    <MyCoursesLessonsStudentItem student={new UserData(1, '12345', 'Davide Falcone', 'davide.falcone@polito.it')} booking={new BookingData(1, '', '12345', 1, 0)} setStudentAsPresent={()=>{}}/>
                </AuthContext.Provider>
            </BrowserRouter>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});