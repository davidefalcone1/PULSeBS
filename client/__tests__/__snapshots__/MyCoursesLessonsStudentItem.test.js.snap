// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyCoursesLessonsStudentItem Renders correctly 1`] = `
<div
  className="list-group-item"
  disabled={false}
  id="student-12345"
  onClick={[Function]}
>
  <div
    className="d-flex w-100 pt-3 justify-content-between no-gutters"
  >
    <div
      className="col-sm-2"
    >
      <p
        id="idOfStudent_12345"
      >
        12345
      </p>
    </div>
    <div
      className="col-sm-3"
    >
      <p
        id="fullNameOfStudent_12345"
      >
        Davide Falcone
      </p>
    </div>
    <div
      className="col-sm-4"
    >
      <p
        id="emailOfStudent12345"
      >
        davide.falcone@polito.it
      </p>
    </div>
    <div
      className="col-sm-3"
    >
      <button
        className="btn btn-success"
        disabled={false}
        id="setPresentOf - 12345"
        onClick={[Function]}
        type="button"
      >
        Set as Present
      </button>
      0
    </div>
  </div>
</div>
`;
